### DISCUSSION TITLE
Certificado A3

### DISCUSSION THREAD

**Discussion by Avelino0neto** [TadaSoftware/PyNFe#268] [Q&A] [answered]
Boa Tarde!

A todos sou novo por aqui tanto no python quanto no projeto, procurei algumas documentações dentro e fora do projeto para tentar fazer o mesmo funcionar com certificado do tipo A3, e não consegui chegar a lugar nenhum será que alguém poderia dar uma luz no fim do túnel, Obrigado pelos criadores do projeto excelente iniciativa.

**Comment by valdecirsanches** [comment]
Seja bem vindo eu ainda estou iniciando.... consegui vencer alguns obstáculos, porem usando somente o A1.... achei mto top consegui aprovar nfe para SP porem estou parado no passo 2 - Cancelar NFe e ai não posso te ajudar muito no seu caso
Hoje eu utilizo acbr com websocket

**Comment by leogregianin** [comment, marked as answer]
@Avelino0neto boa tarde, vai depender se sua aplicação é uma API ou desktop.

Para desktop talvez faça mais sentido atender certificados A3 mas para API não é muito comum, apesar de ser possível com uma interoperabilidade intermediária entre o certificado conectado no desktop com a API.

Nesta biblioteca tem suporte somente a certificados A1. https://github.com/TadaSoftware/PyNFe/blob/d1231c14aa2ff489ba76d4fb5e00ca8863ec1b48/pynfe/entidades/certificado.py#L23

Outra bibliotecas tem esse recursos, exemplo https://github.com/erpbrasil/erpbrasil.assinatura

Qualquer ajuda na implementação disso e de outros recursos é bom vindo.
